estimagic/optimization/transform_problem.py

```diff
@@ -0,0 +1,539 @@
+            Supported keys are:
+                - port (int): port where to display the dashboard
+                - no_browser (bool): whether to display the dashboard in a browser
+                - rollover (int): how many iterations to keep in the monitoring plots
+
+    Returns:
+        optim_kwargs (dict): dictionary collecting the arguments that are going to be
+            passed to _internal_minimize
+        params (pd.DataFrame): The expanded params DataFrame with all needed columns.
+            See :ref:`params`.
+        database_path (str or pathlib.Path or None): path to the database.
+
+    """
+    standard_dash_options = {"no_browser": False, "port": None, "rollover": 500}
+    # important for dash_options to be last for standards to be overwritten
+    dash_options = {**standard_dash_options, **dash_options}
+
+    origin, algo_name = _process_algorithm(algorithm)
+    optim_kwargs = {
+        "origin": origin,
+        "algo_name": algo_name,
+        "algo_options": algo_options,
+    }
+
+    params = _set_params_defaults_if_missing(params)
+    _check_params(params)
+
+    database_path = logging if dashboard else None
+
+    return optim_kwargs, params, dash_options, database_path
+
+
+def _process_algorithm(algorithm):
+    """Identify the algorithm from the user-supplied string.
+
+    Args:
+        algorithm (str): Package and name of the algorithm. It should be of the format
+            {pkg}_{name}.
+
+    Returns:
+        origin (str): Name of the package.
+        algo_name (str): Name of the algorithm.
+
+    """
+    current_dir_path = Path(__file__).resolve().parent
+    with open(current_dir_path / "algo_dict.json") as j:
+        algos = json.load(j)
+
+    origin, algo_name = algorithm.split("_", 1)
+
+    try:
+        assert algo_name in algos[origin], "Invalid algorithm requested: {}".format(
+            algorithm
+        )
+    except (AssertionError, KeyError):
+        proposals = propose_algorithms(algorithm, algos)
+        raise NotImplementedError(
+            f"{algorithm} is not a valid choice. Did you mean one of {proposals}?"
+        )
+
+    return origin, algo_name
+
+
+def _set_params_defaults_if_missing(params):
+    """Set defaults and run checks on the user-supplied params.
+
+    Args:
+        params (pd.DataFrame): See :ref:`params`.
+
+    Returns:
+        params (pd.DataFrame): With defaults expanded params DataFrame.
+
+    """
+    params = params.copy()
+    if "lower" not in params.columns:
+        params["lower"] = -np.inf
+    else:
+        params["lower"].fillna(-np.inf, inplace=True)
+
+    if "upper" not in params.columns:
+        params["upper"] = np.inf
+    else:
+        params["upper"].fillna(np.inf, inplace=True)
+
+    if "group" not in params.columns:
+        params["group"] = "All Parameters"
+
+    if "name" not in params.columns:
+        names = [index_element_to_string(tup) for tup in params.index]
```

Comment:
This function is only used in this module. I would make it a private function in this module.

Reply:
Done in 5933f7dd72674a2ff23e054d995ec2aed33694cc